### PR TITLE
docs:Update the install docs

### PR DESCRIPTION
**Pull request by david-caro**
* Add some missing steps
* Add centos examples
* Add user admin access grant

Signed-off-by: David Caro <david.caro@cern.ch>

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -13,28 +48,42 @@ Create a virtual environment and clone the INSPIRE source code using `git`:
...
 Build assets
 ============
 
 We build assets using `npm`. Make sure you have installed it system wide.
 
+.. note::
+
+    You will have to install a specific version of clean-css until `this issue`_
+    is fixed
+
```
Comment (eamonnmag) [inline INSTALL.rst]: Moving the CSS in to the bundle rather than using as an import solves this, so this problem should now be fixed after the changes made yesterday evening.

Comment (david-caro) [inline INSTALL.rst]: Forgot to remove it :see_no_evil:

File: INSTALL.rst [hunk 2]
```diff
@@ -2,6 +2,41 @@
 Installation
 ============
 
+System prerequisites
+=====================
+
+This guide expects you to have installed is your system the following tools:
```
Comment (jmartinm) [inline INSTALL.rst]: in your system*

File: INSTALL.rst [hunk 3]
```diff
@@ -2,6 +2,41 @@
...
+
+    $ sudo yum install python-pip python-virtualenv python-virtualenvwrapper \
+        npm postgresql postgresql-devel lxml ImageMagick redis 
+    $ sudo npm -g install npm
+
+For elasticsearch you can find the installation instructions on the
+`elasticsearch install page`_, and, to run the development environment, you
+will need also add the following workarounds:
+
+.. code-block:: console
```
Comment (jmartinm) [inline INSTALL.rst]: Note that this is only in CentOS ? At least on MacOS it was not needed, might be good making it clear.

File: INSTALL.rst [hunk 4]
```diff
@@ -175,4 +239,33 @@ entry points and then do the following (e.g. for the `hep` rule):
...
 .. code-block:: console
 
-    dojson -l marcxml -i inspirehep/demosite/data/sample.xml do hep
+    (inspirehep)$ dojson -l marcxml -i inspirehep/demosite/data/sample.xml do hep
+
+
+Giving your user admin access
+==============================
+
+If you want access to the admin console, you'll have to execute:
```
Comment (jmartinm) [inline INSTALL.rst]: Did you create a user after installation?

Note that there is a [fixture](https://github.com/inspirehep/inspire-next/blob/cd50b1cbcd028ba6a00d87d676ed0ed187a77382/inspirehep/modules/fixtures/users.py#L35) that creates an initial user with `superuser` role that should be able to access everything.

So either we mention that the user exists, or we teach how to create a new user (and then add the permissions you mention here)?

File: INSTALL.rst [hunk 5]
```diff
@@ -175,4 +239,33 @@ entry points and then do the following (e.g. for the `hep` rule):
...
+
+    This feature is still unstable, and the name of the authorization access
+    might change, if you are having problems with it, you can check all the
+    access actions with:
+
+    .. code-block:: console
+
+        (inspirehep)$ inspirehep access list
+
+.. _this issue: https://github.com/inspirehep/inspire-next/issues/1296
```
Comment (jmartinm) [inline INSTALL.rst]: What are these links? They don't seem to render in the final docs?

File: INSTALL.rst (hunk 2 above)
Comment (david-caro) [inline INSTALL.rst]: that's generic, you'll need those tools in any system where you plan to run the dev environment, were they installed by default or not. My (CentOS) system specifics are below.

File: INSTALL.rst (hunk 3 above)
Comment (david-caro) [inline INSTALL.rst]: See that at line 21 it states it's an example for CentOS, should I put it in a note block instead?

File: INSTALL.rst (hunk 4 above)
Comment (david-caro) [inline INSTALL.rst]: In the guide it tells you to create that user, I just followed the guide, line 73, has it's own section.

I was unaware of that fixture, is it run by default on install? I can replace the user creation section with this.

File: INSTALL.rst (hunk 5 above)
Comment (david-caro) [inline INSTALL.rst]: They do, just only on pythonhosted (or readthedocs). It's sphinx (restructured text) syntax, and github does not like it, I can try to investigate if it's possible to please both systems in any way

File: INSTALL.rst (hunk 1 above)
Comment (david-caro) [inline INSTALL.rst]: Was that merged? I'm not seeing it in the code, I still see the include in inspirehep/modules/theme/static/scss/holding-pen/holding-pen.scss and It's missing also from the bundles.py file (and it fails to build assets).

Comment (eamonnmag) [inline INSTALL.rst]: We did it on your machine. So it can be your first code contribution 💃

Comment (david-caro) [inline INSTALL.rst]: sent :smile: #1302

**Comment by david-caro** [comment]
I'm retesting with a bare centos docker container, will update if any other package requirements come up

**Comment by david-caro** [comment]
Now it's ready imo

**Comment by david-caro** [comment]
Dammit... no it's not... the todo extension is not added by default to sphinx :/, I'll add it

**Comment by david-caro** [comment]
Done, let me know if you don't want todos there, I can just remove it and leave a comment or something.

File: INSTALL.rst [hunk 6]
```diff
@@ -13,19 +53,58 @@ Create a virtual environment and clone the INSPIRE source code using `git`:
...
+
+    (inspirehep)$ pip install -e .[development]
+
+
+Custom configuration and debug mode
+====================================
+
+If you want to change the database url, or enable the debug mode for
+troubleshooting, you can do so in the `inspirehep.cfg` file under
+`var/inspirehep`, you might need to create it:
```
Comment (jmartinm) [inline INSTALL.rst]: Folder is `var/inspirehep-instance/`

File: INSTALL.rst [hunk 7]
```diff
@@ -13,19 +53,58 @@ Create a virtual environment and clone the INSPIRE source code using `git`:
...
+
+.. code-block:: console
+
+    (inspirehep)$ cdvirtualenv var/inspirehep
+    (inspirehep)$ vim ispirehep.cfg
```
Comment (jmartinm) [inline INSTALL.rst]: :s/ispirehep/inspirehep/g

File: inspirehep/config.py [hunk 8]
```diff
@@ -61,7 +61,7 @@ def _(x):
 # ========
 SQLALCHEMY_DATABASE_URI = os.environ.get(
     "SQLALCHEMY_DATABASE_URI",
-    "postgresql+psycopg2://inspirehep:dbpass123@localhost:5432/inspirehep")
+    "postgresql+psycopg2://inspirehep:dbpass123@172.17.0.2:5432/inspirehep")
```
Comment (jmartinm) [inline inspirehep/config.py]: This should not be commited?

File: INSTALL.rst [hunk 9]
```diff
@@ -34,7 +113,8 @@ We build assets using `npm`. Make sure you have installed it system wide.
 
 .. code-block:: console
 
-    (inspirehep)$ npm update && npm install -g node-sass clean-css requirejs uglify-js
+    (inspirehep)$ sudo npm update
+    (inspirehep)$ sudo npm install -g node-sass clean-css requirejs uglify-js
```
Comment (jacquerie) [inline INSTALL.rst]: FYI `npm install -g` does not require `sudo` using the following configuration: https://docs.npmjs.com/getting-started/fixing-npm-permissions

Comment (david-caro) [inline INSTALL.rst]: That does not look really like 'fixing', it's changing the ownership of a system directory... though the second option there (per-user "global" npm directory is a nice option).

In any case, this is meant to work on a fresh install, with the default permissions. I can add a note pointing to that page if you want. Will have to check if the solution is generic or macos bound tohugh. Btw. there's already a todo message to add ubuntu and macos examples too, might be a subtask of that one

Comment (jacquerie) [inline INSTALL.rst]: Ah, yes, the solution I wanted to point to was the second one, not the ugly `chown`. I _believe_ the official NPM docs discourage using `sudo` in that situation, so I don't think we should recommend that in our installation instructions.

Comment (tiborsimko) [inline INSTALL.rst]: FWIW wouldn't it be easier to lead `npm` installation related discussions in upstream Invenio, and then simply tell the INSPIRE developers (or CDS, or Zenodo...) in the overlay instructions to run the appropriate `provision-web.sh --with-foo --with-bar`, or `provision-elasticsearch.sh` etc scripts, in order to have all the pre-requisites installed? Might be easier to maintain instructions up-to-date in this way.

Comment (david-caro) [inline INSTALL.rst]: I was not aware that those scripts exist, can you point me to them?

Comment (david-caro) [inline INSTALL.rst]: Found them :smile: (thanks @jacquerie), though I don't think it should be added to this pull request right now (it's not just executing a command, but you have to pull in the invenio scripts and setup env vars and such). I think that should go on it's own pr, once we (aka me) have a clear idea of the setup process.

Comment (david-caro) [inline INSTALL.rst]: @jacquerie: I've added a note and a link to that npm docs page, let me know if that's enough (I don't think I should copy the whole stuff, and well, I think they deserve a visit for their info ;) )

File: INSTALL.rst [hunk 10]
```diff
@@ -13,19 +53,58 @@ Create a virtual environment and clone the INSPIRE source code using `git`:
...
 
-Use `pip` to install all requirements:
+Use `pip` to install all requirements, it's recommended to upgrade pip and
+setuptools to latest too:
 
 .. code-block:: console
 
+    (inspirehep)$ pip install --upgrade pip setuptools
     (inspirehep)$ cdvirtualenv src/inspirehep
     (inspirehep)$ pip install -r requirements.txt --pre --exists-action i
```
Comment (jacquerie) [inline INSTALL.rst]: We can probably get rid of the `--exists-action i` now... it's an historical artifact of the Invenio/INSPIRE overlay concept.

File: INSTALL.rst [hunk 11]
```diff
@@ -13,19 +53,58 @@ Create a virtual environment and clone the INSPIRE source code using `git`:
...
 
+And for development:
+
+.. code-block:: console
+
+    (inspirehep)$ pip install -e .[development]
+
+
+Custom configuration and debug mode
+====================================
```
Comment (jacquerie) [inline INSTALL.rst]: Note also that commit 8dacc07d7 introduced another way of overriding configuration via a file called `inspirehep/config_local.py`. I'm not sure it's worth mentioning in the docs, I believe only @eamonnmag uses this variant.

**Comment by jacquerie** [comment]
LGTM. :ship:
